application/forms/Config/UserGroup/AddMemberForm.php

```diff
@@ -133,48 +56,49 @@ public function createElements(array $formData)
      */
     public function onSuccess()
     {
-        $userNames = $this->getValue('user_name') ?: array();
-        if (($users = $this->getValue('users'))) {
-            $userNames = array_merge($userNames, array_map('trim', explode(',', $users)));
-        }
-
-        if (empty($userNames)) {
-            $this->info($this->translate(
-                'Please provide at least one username, either by choosing one '
-                . 'in the list or by manually typing one in the text box below'
-            ));
+        $q = $this->getValue($this->getSearchParameter());
+        if (empty($q)) {
+            Notification::error(t('Please provide at least one username'));
             return false;
         }
 
+        $userNames = array_filter(
+            array_unique(
+                explode(self::TERM_SEPARATOR, urldecode($q))
```

Comment:
Why is `urldecode()` necessary?

Reply:
Because  [BaseInput.js](https://github.com/Icinga/ipl-web/blob/ded069e649f0c4dccca27b36bd6d6d3f34d5bfcb/asset/js/widget/BaseInput.js#L353) encodes the terms.